### PR TITLE
[test] add cmake lists for 143 algorithms

### PR DESCRIPTION
**Pull request by barendgehrels**
This PR
* adds 143 algorithms
* because this caused a duplicate in `perimeter` : it moves those two algorithms to their folder

They all pass on my Mac.
Do we have this CMake flavor on CI already?

### CODE REVIEW

File: test/algorithms/perimeter/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Boost.Geometry
+# Copyright (c) 2024 Barend Gehrels, Amsterdam, the Netherlands.
+# Use, modification and distribution is subject to the Boost Software License,
+# Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+# http://www.boost.org/LICENSE_1_0.txt)
+
+foreach(item IN ITEMS
+    perimeter
+    perimeter_base
```
Comment (barendgehrels) [inline test/algorithms/perimeter/CMakeLists.txt:9]: These two above (were duplicate) are quite similar, and quite short, we can merge them later.
But I wanted this PR to be CMake only.

File: test/algorithms/distance/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    distance_ca_pl_l
+    distance_ca_pl_pl
+    distance_se_geo_ar_ar
+    distance_se_geo_l_ar
+    distance_se_geo_l_l
+    distance_se_geo_pl_ar
+    distance_geo_pl_l
+    distance_se_geo_pl_pl
+    distance_se_pl_l
+    distance_se_pl_pl
```
Comment (barendgehrels) [inline test/algorithms/distance/CMakeLists.txt:22]: All these lists are made by using relevant lines from the `Jamfile` and stripping all but the filename.

File: test/algorithms/closest_points/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+foreach(item IN ITEMS
+    ar_ar
+    l_ar
+    l_l
+    pl_ar
+    pl_l
+    pl_pl
+    )
+  boost_geometry_add_unit_test("algorithms_closest_points" ${item})
```
Comment (barendgehrels) [inline test/algorithms/closest_points/CMakeLists.txt:15]: This one is the only suffixing the name itself to the algorithms, for obvious reasons

**Review by vissarion — APPROVED** [review]
Thanks, I think it is very useful to have a complete cmake support for tests. 

Regarding perimeter tests, should you also move `test_perimeter.hpp` in `perimeter` directory? 

There is no cmake related testing in CI, we should include it. See also https://github.com/boostorg/geometry/pull/1273#issuecomment-2102961622

**Comment by barendgehrels** [comment]
> Thanks, I think it is very useful to have a complete cmake support for tests.

The algorithms should be complete now. But there is still some work to do for several folders (among others: `policies` , `iterators`, `formulas`)

**Comment by barendgehrels** [comment]
> There is no cmake related testing in CI, we should include it. See also [#1273 (comment)](https://github.com/boostorg/geometry/pull/1273#issuecomment-2102961622)

Thanks, looking forward to it.

**Comment by barendgehrels** [comment]
@vissarion 
I automated it a bit (not checked in) with help of ChatGPT, last pushes were from that effort. That also adds `detail/sections` which I missed before.

Next PR will add the next list, and then it should be complete.
```
add_subdirectory(core)
add_subdirectory(concepts)
add_subdirectory(geometries)
add_subdirectory(arithmetic)
add_subdirectory(formulas)
add_subdirectory(iterators)
add_subdirectory(strategies)
add_subdirectory(policies)
add_subdirectory(io)
add_subdirectory(srs)
```

**Pull request merged**